### PR TITLE
Migrated the database to fix label names and support trash date of emails.

### PR DESCRIPTION
**Pull request by jorgeblacio**
- Also added a new table to handle the resend of emails when the email has non criptext recipients and the email was encrypted.

### CODE REVIEW

File: src/main/kotlin/com/criptext/mail/SecureEmail.kt [hunk 1]
```diff
@@ -26,16 +26,16 @@ class SecureEmail {
...
+        val LABEL_ALL_MAIL = "All Mail"
         val LABEL_CATEGORY_PROMOTIONS = "CATEGORY_PROMOTIONS"
-        val LABEL_DRAFT = "DRAFT"
-        val LABEL_SPAM = "SPAM"
-        val LABEL_STARRED = "STARRED"
-        val LABEL_UNREAD = "UNREAD"
+        val LABEL_DRAFT = "Draft"
+        val LABEL_SPAM = "Spam"
+        val LABEL_STARRED = "Starred"
+        val LABEL_UNREAD = "Unread"
```
Comment (not-gabriel) [inline src/main/kotlin/com/criptext/mail/SecureEmail.kt]: Why use this instead of `Label.DefaultItems`?

Comment (jorgeblacio) [reply src/main/kotlin/com/criptext/mail/SecureEmail.kt]: Now there are constants for the labels inside the label model companion object and the default label uses those instead, as well as all the code that needed to check for one of these.

Comment (danieltigse) [reply src/main/kotlin/com/criptext/mail/SecureEmail.kt]: I would suggest to delete `SecureEmail.kt`

File: src/main/kotlin/com/criptext/mail/db/AppDatabase.kt [hunk 2]
```diff
@@ -59,10 +65,32 @@ abstract class AppDatabase : RoomDatabase() {
...
                         "encriptedMail1")
                         //allowMainThreadQueries() // remove this in production... !!!!
-                        // .addMigrations(MIGRATION_1_2, MIGRATION_2_3, MIGRATION_1_3)
+                        .addMigrations(MIGRATION_1_2)
                         .build()
             }
             return INSTANCE!!
         }
+
+        val MIGRATION_1_2: Migration = object : Migration(1, 2) {
```
Comment (danieltigse) [inline src/main/kotlin/com/criptext/mail/db/AppDatabase.kt]: add the code to "fix" the contacts

**Pull request merged**